### PR TITLE
Don't hardcode barrel shapes

### PR DESCRIPTION
**Pull request by Craluminum2413**

### CODE REVIEW

**Comment by Guimoute** [comment]
It's bad practice to copy paste

image

5 times. You can clearly see the pattern to write the static stuff only once.

**Comment by Craluminum2413** [comment]
> It's bad practice to copy paste
> 
> image
> 
> 5 times. You can clearly see the pattern to write the static stuff only once.

Do better then

**Comment by Craluminum2413** [comment]
@Guimoute do better then

File: Systems/Barrel/BlockBarrel.cs [hunk 1]
```diff
@@ -345,7 +350,31 @@ public override void OnLoaded(ICoreAPI api)
...
+            }
+            
+            if (Attributes?["contentsShape"].Exists == true)
+            {
+                contentsShape = AssetLocation.Create(Attributes["contentsShape"].AsString("shapes/block/wood/barrel/contents.json")).WithPathPrefixOnce("shapes").WithPathAppendixOnce(".json");
+            }
+                        
+            if (Attributes?["opaqueLiquidContentsShape"].Exists == true)
+            {
+                contentsShape = AssetLocation.Create(Attributes["opaqueLiquidContentsShape"].AsString("shapes/block/wood/barrel/opaqueliquidcontents.json")).WithPathPrefixOnce("shapes").WithPathAppendixOnce(".json");
```
Comment (tyronx) [inline Systems/Barrel/BlockBarrel.cs:370]: wrong field name here, and this code is indeed a bit suboptimal. will fix.

Comment (tyronx) [reply Systems/Barrel/BlockBarrel.cs:370]: domain field also not correctly set

Comment (tyronx) [reply Systems/Barrel/BlockBarrel.cs:370]: Correct solution
image

**Comment by tyronx** [comment]
this is now merged

**Pull request closed without merging**